### ISSUE TITLE
Ability to get list of roles

### ISSUE THREAD

**Issue by lordekar** [OptimalBits/node_acl#133]
Hello, I'm working on writing simple forum system and got issue with roles. The forum has to has ability to edit existing roles and setup new ones. I looked up through source code and didn't find any function or storage  of roles list. It seems to roles are storing only in backend.

```javascript
acl.getRoles(function(err, roles){
...
})
```

Sorry for my English

**Comment by ErikvdVen** [comment]
Same question over here, would be a nice feature!

**Comment by jonmassot** [comment]
IMO, I don't think it was expected to work like that. Since all the users data is not to be kept into ACL database, roles should be the same. You can always write a `plugin` over node-acl that implement that kind of functionality.

**Comment by ErikvdVen** [comment]
Well, for me it's handy to have a function to return a list of all Roles (from the meta bucket actually, cause it's the only bucket which contains all roles, even if the role doesn't contain any users yet). 

The reason is, we would like to create a dashboard which allows Admin users to manage all roles and permissions. And to easily add a new user to a specific role with specific rights, through an interface.

**Comment by jonmassot** [comment]
@ErikvdVen Yes, that's what I'm saying. I pretty much understand the use case, I'm just arguing as to why you want to have roles (without users) in the `acl` database that should be more like a collection of authorization and not a collection of roles and authorization.

**Comment by ErikvdVen** [comment]
Alright! Thanks for thinking along! We just started creating a separated collection with all roles  :+1:

**Comment by jonmassot** [comment]
Cool. Glad I could be of any help!

**Comment by trainerbill** [comment]
I just ran into this issue.  say I am creating an admin section that defines role permissions.  I need to be able to list all the roles in the acl_roles collection.  Currently that is not possible without queying the collection independently...  which means I will probably need a mongoose schema.  Not sure why there is not basic CRUD functionality on this module.

**Comment by anandka** [comment]
Hey trainerbill any updates of how to do it without mongoose schema?

**Comment by deksden** [comment]
@trainerbill : You can use roles same as users: just ids of them, not whole objects. 

Store Roles, Users outside of ACL storage, and use ACL only for defining relationships between permissions/roles/users.
